.github/workflows/e2e-agora.yaml

```diff
@@ -53,49 +47,77 @@ jobs:
           done
           echo "AFFECTED=false" >> "${GITHUB_OUTPUT}"
 
+      - name: Remove the dev container
+        run: docker rm -f sage_devcontainer
+  run-agora-e2e-tests:
+    needs: check-agora-affected
+    if: needs.check-agora-affected.outputs.agora_affected == 'true'
+    environment: ${{ github.event_name == 'pull_request_target' &&
```

Comment:
Could you please add a comment on top to describe the behavior of this `environment` expression? This should make this bit of code more understandable and hence reduce the chances of bugs.